testbed/tests/window/test_window.py

```diff
@@ -555,13 +657,236 @@ async def test_full_screen(second_window, second_window_probe):
         assert second_window_probe.is_resizable
         assert second_window_probe.content_size == initial_content_size
 
-        second_window.full_screen = False
+        with pytest.warns(
+            DeprecationWarning,
+            match="`Window.full_screen` is deprecated. Use `Window.state` instead.",
+        ):
+            second_window.full_screen = False
         await second_window_probe.wait_for_window(
             "Secondary window is still not full screen"
         )
         assert not second_window_probe.is_full_screen
         assert second_window_probe.content_size == initial_content_size
 
+    @pytest.mark.parametrize(
```

Comment:
This set of new tests is essentially only testing the "simple" case - NORMAL -> (some other state). They also need to validate the other transitions e.g., MINIMIZED->MAXIMIZED, MINIMIZED->FULLSCREEN, MINIMIZED->PRESENTATION MODE - because these are the cases that are complex on a per-backend basis.

Reply:
I have added a new test.